### PR TITLE
[SASS-6247] Upgrade Expenses journeys - 2

### PR DESCRIPTION
**Pull request by Jake-Raffe**

### CODE REVIEW

File: test/services/SelfEmploymentServiceSpec.scala [hunk 1]
```diff
@@ -178,39 +188,94 @@ class SelfEmploymentServiceSpec extends SpecBase with MockitoSugar with Argument
...
+            taxYear,
+            NormalMode)
+          .futureValue
+
+      val expectedAnswers = existingZegvAnswers.set(ZeroEmissionGoodsVehiclePage, false, businessId.some).success.value
+      val dbAnswers       = repository.state(userAnswersId)
       assert(dbAnswers === expectedAnswers)
+
+      assert(result.header.status === SEE_OTHER)
+      assert(result.header.headers.head._2 === ZeroEmissionGoodsVehiclePage.cyaPage(taxYear, businessId).url)
```
Comment (harry-ritchie) [inline test/services/SelfEmploymentServiceSpec.scala:251]: We could use Play's built in helpers for this, where we pass in in the future:
```scala
status(result) shouldBe ...
redirectLocation(result) shouldBe ...
```

File: app/controllers/journeys/expenses/goodsToSellOrUse/DisallowableGoodsToSellOrUseAmountController.scala [hunk 2]
```diff
@@ -17,67 +17,59 @@
...
+  private val page = DisallowableGoodsToSellOrUseAmountPage
 
-          Future.successful(Ok(view(preparedForm, mode, request.userType, taxYear, businessId, formatMoney(goodsAmount))))
-      }
+  def onPageLoad(taxYear: TaxYear, businessId: BusinessId, mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData) {
+    implicit request =>
+      (for {
+        allowableAmount <- request.valueOrRedirectDefault(GoodsToSellOrUseAmountPage, businessId)
+        form = fillForm(page, businessId, formProvider(request.userType, allowableAmount))
+      } yield Ok(view(form, mode, request.userType, taxYear, businessId, formatMoney(allowableAmount)))).merge
```
Comment (harry-ritchie) [inline app/controllers/journeys/expenses/goodsToSellOrUse/DisallowableGoodsToSellOrUseAmountController.scala:55]: Not wrong by any means, but worth considering just explicitly using map if we are not sequencing any monadic operations.

File: test/controllers/actions/SubmittedDataRetrievalActionProviderImplSpec.scala [hunk 3]
```diff
@@ -55,7 +56,7 @@ class SubmittedDataRetrievalActionProviderImplSpec extends AnyWordSpecLike with
 
       val underTest = new SubmittedDataRetrievalActionProviderImpl(connector, repo)
 
-      val result = underTest.loadTaskList(taxYear, fakeOptionalRequest).value.futureValue
```
Comment (harry-ritchie) [inline test/controllers/actions/SubmittedDataRetrievalActionProviderImplSpec.scala:58]: Just out-of-interest, did you replace `futureValue` with `await` as it gave you a future not completed test failure?

File: test/services/SelfEmploymentServiceSpec.scala (hunk 1 above)
Comment (Jake-Raffe) [reply test/services/SelfEmploymentServiceSpec.scala:251]: agreed, thanks

File: test/controllers/actions/SubmittedDataRetrievalActionProviderImplSpec.scala (hunk 3 above)
Comment (Jake-Raffe) [reply test/controllers/actions/SubmittedDataRetrievalActionProviderImplSpec.scala:58]: Yes, the test would start failing half the time from a timeout

File: app/controllers/journeys/expenses/goodsToSellOrUse/DisallowableGoodsToSellOrUseAmountController.scala (hunk 2 above)
Comment (Jake-Raffe) [reply app/controllers/journeys/expenses/goodsToSellOrUse/DisallowableGoodsToSellOrUseAmountController.scala:55]: Yep, think you're right. Will change them

**Review by harry-ritchie — APPROVED** [review]

**Pull request merged**